xrmocap/human_perception/keypoints_estimation/mediapipe_estimator.py

```diff
@@ -0,0 +1,300 @@
+                    It could be smaller than the input bbox_list,
+                    if there's no keypoints detected in some bbox.
+        """
+        ret_kps_list = []
+        ret_boox_list = []
+        if load_batch_size is None:
+            load_batch_size = len(frame_path_list)
+        for start_idx in range(0, len(frame_path_list), load_batch_size):
+            end_idx = min(len(frame_path_list), start_idx + load_batch_size)
+            if load_batch_size < len(frame_path_list):
+                self.logger.info(
+                    'Processing mediapipe on frames' +
+                    f'({start_idx}-{end_idx})/{len(frame_path_list)}')
+            image_array_list = []
+            for frame_abs_path in frame_path_list[start_idx:end_idx]:
+                img_np = cv2.imread(frame_abs_path)
+                image_array_list.append(img_np)
+            batch_pose_list, _, batch_boox_list = \
+                self.infer_array(
+                    image_array=image_array_list,
+                    bbox_list=bbox_list[start_idx:end_idx],
+                    disable_tqdm=disable_tqdm)
+            ret_kps_list += batch_pose_list
+            ret_boox_list += batch_boox_list
+        return ret_kps_list, None, ret_boox_list
+
+    def infer_video(self,
+                    video_path: str,
+                    bbox_list: Union[tuple, list],
+                    disable_tqdm: bool = False,
+                    **kwargs) -> Tuple[list, list]:
+        """Infer frames from a video file.
+
+        Args:
+            video_path (str):
+                Path to the video to be detected.
+            bbox_list (Union[tuple, list]):
+                A list of human bboxes.
+                Shape of the nested lists is (n_frame, n_human, 5).
+                Each bbox is a bbox_xyxy with a bbox_score at last.
+            disable_tqdm (bool, optional):
+                Whether to disable the entire progressbar wrapper.
+                Defaults to False.
+
+        Returns:
+            Tuple[list, list]:
+                keypoints_list (list):
+                    A list of human keypoints.
+                    Shape of the nested lists is
+                    (n_frame, n_human, n_keypoints, 3).
+                    Each keypoint is an array of (x, y, confidence).
+                bbox_list (list):
+                    A list of human bboxes.
+                    Shape of the nested lists is (n_frame, n_human, 5).
+                    Each bbox is a bbox_xyxy with a bbox_score at last.
+                    It could be smaller than the input bbox_list,
+                    if there's no keypoints detected in some bbox.
+        """
+        image_array = video_to_array(input_path=video_path, logger=self.logger)
+        ret_kps_list, _, ret_boox_list = self.infer_array(
+            image_array=image_array,
+            bbox_list=bbox_list,
+            disable_tqdm=disable_tqdm)
+        return ret_kps_list, None, ret_boox_list
```

Comment:
For `infer_video`, `infer_frames` and `infer_array`, why we return 3 values and the middle one is `None`?

Reply:
This is for aligning the function of `MMposeTopDownEstimator`.  `MMposeTopDownEstimator` returns one more value of heatmaps.